### ISSUE TITLE
send object to log change the object 

### ISSUE THREAD

**Issue by dtagliavini** [mperdeck/jsnlog#153]
Hi,

I'm using an external library that use it's method to log to console in this way:
```
Janus.debug = console.debug.bind(console);
Janus.log = console.log.bind(console);
....
```
To send this message to jsnlog and indeed to consoleappender or/and ajax appender I bind these functons in this way:
```
Janus.debug = JL('JANUS').debug.bind(JL('JANUS'));
```

And this seems to works fine with message string, so a Janus.debug('Everythings is ok") go to jsnlog.

But a call like this creates me problem:

```
var pc_constraints = {
	"optional": [{"DtlsSrtpKeyAgreement": true}]
};
......
......
Janus.debug(pc_constraints);
config.pc = new RTCPeerConnection(pc_config, pc_constraints);
```
this because in this function pc_constraints is changed in this way.
```
Logger.prototype.log = function (level, logObject, e) {
........
........
                    compositeMessage.meta = compositeMessage.meta || {};
                    compositeMessage.meta.loggerName = this.loggerName;
```

After this call my object results in:
```
{
	"optional": [{"DtlsSrtpKeyAgreement": true}]
	"loggerName": "JANUS"
};
```
and this call fails:
```
config.pc = new RTCPeerConnection(pc_config, pc_constraints);
```
with a: Failed to construct 'RTCPeerConnection': Malformed constraints object.

Is this a issue or I'm binding in the wrong way?

Thanks in advance.

**Comment by mperdeck** [comment]
It seems to me that this is a Janus issue, not a JSNLog issue. I'm not familiar with Janus.

It seems to me that the bind is used to tell Janus how to log something (via the console, via JSNLog, etc.) I guess that you're passing an object or function to it. And that the Janus documentation shows what that object or function should look like. You could create your own object or function that adheres to Janus' requirements, and then have your object or function call JSNLog.

**Comment by dtagliavini** [comment]
I think you didn't understand me.

In your document you explain that it's possible to log object in this way:

```
JL().fatal({"i": 5, "j": 6});
```

This is my code to reproduce what I mean [(jsfiddle demo)](https://jsfiddle.net/j8wgm3hp/) :

```
var my= {"i": 5, "j": 6};
alert(JSON.stringify(my));
JL().fatal(my);
alert(JSON.stringify(my));
```

The second alert show you that the object my is changed.

**Comment by mperdeck** [comment]
I understand now. Yes you're right. It will be fixed in the next release 2.24.3.
